### PR TITLE
Exit node with programResult

### PR DESCRIPTION
**Pull request by alehander42**
programResult wasn't being honored by node backend.

I am not sure if there are tests for exit code?

### CODE REVIEW

**Comment by alehander42** [comment]
Actually nvm, there is a more correct solution: I'll fix the PR

**Comment by alehander42** [comment]
Ok, this should work.

File: lib/system.nim [hunk 1]
```diff
@@ -1439,7 +1439,11 @@ const
     ## is the value that should be passed to `quit <#quit>`_ to indicate
     ## failure.
 
-var programResult* {.exportc: "nim_program_result".}: int
+when defined(nodejs):
+  var programResult* {.importc: "process.exitCode".}: int
+  programResult = 0
```
Comment (dom96) [inline lib/system.nim]: Why is this necessary?

Comment (alehander42) [reply lib/system.nim]: because it's undefined by default and that breaks +=
